kafka-connector/src/test/java/com/google/pubsub/kafka/sink/CloudPubSubSinkTaskTest.java

```diff
@@ -173,12 +173,52 @@ public void testStructSchema() {
     List<SinkRecord> list = new ArrayList<>();
     list.add(record);
     task.put(list);
-    schema = SchemaBuilder.struct().field(FIELD_STRING1, SchemaBuilder.struct()).build();
-    record = new SinkRecord(null, -1, null, null, schema, new Struct(schema), -1);
+  }
+
+  @Test
+  public void testStructSchemaWithOptionalField() {
+    task.start(props);
+
+    Schema schema = SchemaBuilder.struct().field(FIELD_STRING1, SchemaBuilder.string())
+        .field(FIELD_STRING2, SchemaBuilder.string().optional()).build();
+    Struct val = new Struct(schema);
+    val.put(FIELD_STRING1, "tide");
+    SinkRecord record = new SinkRecord(null, -1, null, null, schema, val, -1);
+    List<SinkRecord> list = new ArrayList<>();
     list.add(record);
-    try {
-      task.put(list);
-    } catch (DataException e) { } // Expected, pass.
+    task.put(list);
```

Comment:
Could you add a case where the optional field is filled? I think it could be in this test method, just cover both cases with optional types.

Reply:
Done.